scripts/linters/pylint_extensions.py

```diff
@@ -1960,6 +2029,398 @@ def process_tokens(self, tokens):
+                import_status_dict['single_line_import'] = True
+                import_status_dict['import_line_num'] = line_num
+
+        # Checking if multi-line import is present.
+        # Eg: from typing import (
+        #   Any, Callable, Dict, FrozenSet, Iterator, List, Set,
+        #   Tuple, Type, cast
+        # )
+        if token_type == tokenize.OP:
+            if import_status_dict['single_line_import'] and token == '(':
+                import_status_dict['inside_multi_line_import_scope'] = True
+                import_status_dict['single_line_import'] = False
+            if (
+                import_status_dict['inside_multi_line_import_scope'] and
+                token == ')'
+            ):
+                import_status_dict['inside_multi_line_import_scope'] = False
+
+    def _check_exceptional_type_is_documented(
+        self, type_status_dict, import_status_dict, token_type, token,
+        line, line_num, exceptional_type
+    ):
+        """Checks whether the given exceptional type in a module has been
+        documented or not. If the exceptional type is not documented then
+        adds an error message.
+
+        Args:
+            type_status_dict: dict. The dict containing all the information
+                about the exceptional type that was passed to this method.
+            import_status_dict: Optional[Dict]. This dictionary contains the
+                variables that tracks the module's import status, whether a
+                multi-line import or single-line import is present, or None
+                if the given exceptional_type is not imported in the module.
+            token_type: int. The kind of token that pylint provided.
+            token: str. The token of module the pylint provided.
+            line: str. The line of the module where current token is present.
+            line_num: int. The line number of given token.
+            exceptional_type: str. The exceptional type for which this method
+                is called, Possible values can be 'Any' or 'object'.
+        """
+        # Here, we are defining new variables so that we don't have to fetch
+        # the values from 'type_status_dict' dictionary again and again.
+        outside_function_signature_block = type_status_dict[
+            'outside_function_signature_block'
+        ]
+        outside_args_section = type_status_dict['outside_args_section']
+        args_section_end_line_num = type_status_dict[
+            'args_section_end_line_num'
+        ]
+        type_present_inside_arg_section = type_status_dict[
+            'type_present_inside_arg_section'
+        ]
+        type_present_inside_return_section = type_status_dict[
+            'type_present_inside_return_section'
+        ]
+        type_present_in_function_signature = type_status_dict[
+            'type_present_in_function_signature'
+        ]
+        type_comment_present = type_status_dict['type_comment_present']
```

Comment:
I have tried this approach and after this approach, the code looks more complicated because the dictionary name and variable names are long so they usually exceed the 80 char limit. Also, we are accessing the same variable sometimes so I think it is better to use a variable instead of accessing Dict again and again.

Reply:
Can you shorten the variable/key names? I agree that unpacking and repacking the dictionary seems overly complicated

Reply:
Now we are using dict, variables are removed!